pkg/vm/engine/disttae/types.go

```diff
@@ -286,6 +307,24 @@ func (cols Columns) Len() int           { return len(cols) }
 func (cols Columns) Swap(i, j int)      { cols[i], cols[j] = cols[j], cols[i] }
 func (cols Columns) Less(i, j int) bool { return cols[i].num < cols[j].num }
 
+func (es entries) Len() int      { return len(es) }
+func (es entries) Swap(i, j int) { es[i], es[j] = es[j], es[i] }
+func (es entries) Less(i, j int) bool {
+	if es[i].key < es[j].key {
+		return true
+	}
+	if es[i].key > es[j].key {
+		return false
+	}
+	if es[i].ts.Less(es[j].ts) {
+		return true
+	}
+	if es[i].ts.Greater(es[j].ts) {
+		return false
+	}
+	return es[i].val == nil && es[j].val != nil
```

Comment:
tae's log order is disordered, so i need to put the same timestamp of delete in front of insert.